packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -425,8 +438,8 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, Semver {
             })
         );
 
-        // Set the L1 head hash at the time of the game's creation.
-        l1Head = Hash.wrap(blockhash(block.number - 1));
+        // Set the L1 head hash to the block hash of the L1 block number provided.
+        l1Head = BLOCK_HASH_ORACLE.load(_getArgUint256(0x40));
```

Comment:
What happens if this reverts?

Reply:
Purposefully don't handle reverts gracefully here, a game needs the L1 block hash. This revert will bubble up to the `DisputeGameFactory` and the game won't be created.

Reply:
Do you mind adding this as a comment?